lib/Infer/AbstractInterpreter.cpp

```diff
@@ -144,6 +144,25 @@ namespace souper {
     return S;
   }
 
+  llvm::KnownBits concatKnownBits(llvm::KnownBits A, llvm::KnownBits B) {
+    auto W0 = A.Zero.getBitWidth();
+    auto W1 = B.Zero.getBitWidth();
+
+    llvm::KnownBits Result(W0 + W1);
+    Result.Zero = 0;
+    Result.One = 0;
+
+    Result.Zero |= A.Zero;
```

Comment:
want to make this a little simpler? instead of initialize to zero and then OR it with some stuff, just initialize with the stuff you want there in the first place

Reply:
done